src/components/VmActions/index.js

```diff
@@ -91,76 +92,154 @@ EmptyAction.propTypes = {
  * Active actions on a single VM-card.
  * List of actions depends on the VM state.
  */
-const VmActions = ({ vm, isOnCard = false, onGetConsole, onShutdown, onRestart, onForceShutdown, onStart, onSuspend }) => {
-  const status = vm.get('status')
-
-  const confirmShutdown = (e) => OnClickTopConfirmation({
-    id: vm.get('id'),
-    target: e.target,
-    confirmationText: 'Shut down the VM?',
-    cancelLabel: 'Cancel',
-    okLabel: 'Yes',
-    extraButtonLabel: 'Force',
-    onOk: onShutdown,
-    onExtra: onForceShutdown,
-  })
+class VmActions extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      openModal: false,
+      notificationTarget: null,
+    }
+    this.onConsoleClick = this.onConsoleClick.bind(this)
+    this.onModalClose = this.onModalClose.bind(this)
+    this.onConsoleDownload = this.onConsoleDownload.bind(this)
+    this.confirmShutdown = this.confirmShutdown.bind(this)
+    this.confirmRestart = this.confirmRestart.bind(this)
+    this.confirmSuspend = this.confirmSuspend.bind(this)
+  }
 
-  const confirmRestart = (e) => OnClickTopConfirmation({
-    id: vm.get('id'),
-    target: e.target,
-    confirmationText: 'Restart the VM?',
-    cancelLabel: 'Cancel',
-    okLabel: 'Yes',
-    onOk: onRestart,
-  })
+  onConsoleClick (e) {
+    this.setState({
+      openModal: true,
+      notificationTarget: e.target,
+    })
+    this.props.onCheckConsoleSessionInUse()
+  }
 
-  const confirmSuspend = (e) => OnClickTopConfirmation({
-    id: vm.get('id'),
-    target: e.target,
-    confirmationText: 'Suspend the VM?',
-    cancelLabel: 'Cancel',
-    okLabel: 'Yes',
-    onOk: onSuspend,
-  })
+  onModalClose () {
+    this.setState({
+      openModal: false,
+      notificationTarget: null,
+    })
+    this.props.onConsoleSessionConfirmaClose()
+  }
+
+  onConsoleDownload () {
+    this.setState({
+      openModal: false,
+      notificationTarget: null,
+    })
+
+    this.props.onDownloadConsole()
+  }
+
+  componentDidUpdate (prevProps, prevState) {
+    if (
+        this.props.VmAction.getIn(['vms', this.props.vm.get('id'), 'consoleInUse']) === true &&
+        this.state.openModal === true
+      ) {
+      (() => OnClickTopConfirmation({
+        id: this.props.vm.get('id'),
+        target: this.state.notificationTarget,
+        confirmationText: 'Console in use, continue?',
+        cancelLabel: 'Cancel',
+        okLabel: 'Yes',
+        onOk: this.onConsoleDownload,
+        onCancel: this.onModalClose,
+      }))()
```

Comment:
Can be the OnClickTopConfirmation() function executed directly without the wrapping?